src/cobertura.d

```diff
@@ -0,0 +1,199 @@
+module cobertura;
+
+import std.typecons : Nullable, nullable;
+import std.traits : isSomeString, isCallable;
+import std.range : isInputRange, ElementType;
+
+struct LineCoverage {
+	size_t index;
+	size_t count;
+}
+
+struct FileCoverage {
+	string path;
+	LineCoverage[] lines;
+	size_t linesCovered;
+}
+
+FileCoverage parseLstFile(string path) {
+	import std.stdio : File;
+	return parseLstFile(File(path).byLine, path);
+}
+
+template tryTo(T) {
+	Nullable!T tryTo(R)(R r) {
+		import std.conv : to;
+		try {
+			return nullable(r.to!T);
+		} catch (Exception e) {
+			return Nullable!T.init;
+		}
+	}
+}
+
+enum isNullable(T) = is(T : Nullable!P, P);
+
+auto filterNullable(Range)(Range r) if (isNullable!(ElementType!Range)) {
+	import std.algorithm : filter, map;
+	return r.filter!(a => !a.isNull).map!(a => a.get);
+}
+
+private template andThen(alias fun) {
+    auto andThen(T)(Nullable!T t) {
+        alias RT = typeof(fun(T.init));
+        static if (is(RT == void)) {
+            if (!t.isNull)
```

Comment:
this is a nitpick but, if statements should always be followed by block statements. Single statement version will always bite somebody down the road

Reply:
Sure.